data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java

```diff
@@ -173,48 +181,36 @@ public void createInputFile() throws IOException {
   }
 
   public void createOrcInputFile(List<Record> records) throws IOException {
-    if (ORC_FILE.exists()) {
-      Assert.assertTrue(ORC_FILE.delete());
-    }
```

Comment:
From my experience with this, the file won't write if it already exists.

So you can either create a directory and then create a file underneath to write to, or to keep things simple and just keep the present exists / delete (which will still ensure that the file clean up happens properly).